pygeo/geo_utils/polygon.py

```diff
@@ -1,15 +1,39 @@
 import numpy as np
+from . import euclideanNorm
 
 # --------------------------------------------------------------
 #                Polygon geometric functions
 # --------------------------------------------------------------
 
 
+def areaTri(p0, p1, p2):
+    """
+    compute area based on three point arrays
+    """
+    # convert p1 and p2 to v1 and v2
+    v1 = p1 - p0
+    v2 = p2 - p0
+
+    # compute the areas
+    areaVec = np.cross(v1, v2)
+
+    # area = np.linalg.norm(areaVec,axis=1)
+    area = 0
+    for i in range(len(areaVec)):
+        area += euclideanNorm(areaVec[i, :])
+
+    # return np.sum(area)/2.0
+    return area / 2.0
+
+
 def areaPoly(nodes):
-    # Return the area of the polygon. Note that the input need not be
-    # strictly a polygon, (closed curve in 2 dimensions.) The approach
-    # we take here is to find the centroid, then sum the area of the
-    # 3d triangles. THIS APPROACH ONLY WORKS FOR CONVEX POLYGONS!
+    """Return the area of the polygon.
+    Note that the input need not be strictly a polygon (closed curve in 2 dimensions).
+    The approach we take here is to find the centroid, then sum the area of the
+    3d triangles.
+
+    .. warning:: This approach only works for convex polygons
```

Comment:
I'm not sure I understand why this is. It is not necessary, but if you have additional insight could you add it to the warning.

Reply:
I just converted the previous block comment to docstrings, and changed the all caps warning into a warning block. I'm not sure whether the warning is actually true or not, but if you think it is not the case then I can remove it.

Reply:
I don't know if it true, so i guess it is best to leave it be.